aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs

```diff
@@ -11,8 +11,39 @@ namespace Entities.Weapons
     /// </summary>
     public class RangedWeapon : Weapon
     {
-        [SerializeField] private int _damage = 50;
-        [SerializeField] private int _range = 50;
+        /// <summary>
+        /// The point from which the weapon will be fired.
+        /// </summary>
+        [SerializeField]
+        private Transform _firePoint;
+
+        [SerializeField]
+        private int _range = 50;
+
+        private IEnumerable<RaycastHit> _orderedHits;
+
+        /// <summary>
+        /// By default, assume the weapon will be fired by the player, from the camera.
+        /// </summary>
+        private void Awake()
+        {
+            if (_firePoint == null)
```

Comment:
Might want to do the actual null-check here? But I might be wrong.
```suggestion
            if (_firePoint is null)
```